### DISCUSSION TITLE
Git-submodules: cannot be used on repositories using https for something else

### DISCUSSION THREAD

**Discussion by loikki** [renovatebot/renovate#28872] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Renovate on k8s with helm chart 37.318.1 from https://docs.renovatebot.com/helm-charts

### Please tell us more about your question or problem

Hi,
I am pretty sure that renovate will speedup my work a lot so thanks a lot for your work. Unfortunately, I am struggling with a bug.

I am trying to keep a submodule up to date with the latest tag but renovate cannot reach the [git repository](https://git.whamcloud.com/fs/lustre-release.git).

In my understanding, the issue comes from the fact that renovate is [replacing the protocol by https](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/git-submodules/extract.ts#L129). Unfortunately the website does not provide git through the https port only through the git one. So I would need renovate to use either git://git.whamcloud.com/fs/lustre-release.git or ssh with git@git.whamcloud.com/fs/lustre-release.

I don't see any option to prevent renovate to make the change. Did I miss something or does renovate need a fix?

### Logs (if relevant)

<details><summary>Logs</summary>

```
WARN: Error mapping git submodules during extraction (repository=hpc/cdist-conf, packageFile=.gitmodules)
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "--symref",
             "https://git.whamcloud.com/fs/lustre-release",
             "HEAD"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "fatal: repository 'https://git.whamcloud.com/fs/lustre-release/' not found\n",
         "stack": "Error: fatal: repository 'https://**redacted**@3.24.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/usr/local/renovate/node_modules/.pnpm/simple-git@3.24.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:54:29)\n    at /usr/local/renovate/node_modules/.pnpm/simple-git@3.24.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/local/renovate/node_modules/.pnpm/simple-git@3.24.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:121:14)\n    at GitExecutorChain.<anonymous> (/usr/local/renovate/node_modules/.pnpm/simple-git@3.24.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:97:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/local/renovate/node_modules/.pnpm/simple-git@3.24.0/node_modules/simple-git/dist/cjs/index.js:52:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
       }


```

</details>

**Comment by rarkins** [comment]
Updating git-submodules over git/ssh is not supported yet. The git-submodules manager is classified as in beta. PRs are welcome either to improve the docs in that area or implement the feature

**Comment by loikki** [comment, marked as answer]
Thanks for the quick answer.

I have decided to use a mirror on gitlab and it works very well!

**Reply by rarkins** [reply]
Thanks for sharing your workaround!
